public/app/plugins/datasource/tempo/datasource.ts

```diff
@@ -204,9 +204,21 @@ export class TempoDatasource extends DataSourceWithBackend<TempoQuery, TempoJson
       options = await this.languageProvider.getOptionsV1(labelName);
     }
 
-    return options.filter((option) => option.value !== undefined).map((option) => ({ text: option.value })) as Array<{
-      text: string;
-    }>;
+    // Transform and filter options.
+    // We do not use `filter` and `map`, e.g.,
+    // ```
+    // options.filter((option) => option.value !== undefined).map((option) => ({ text: option.value }))
+    // ```
+    // because TypeScript cannot properly infer types with them and a type error would be raised.
+    const parsedOptions: Array<{ text: string }> = [];
+    options.forEach((option) => {
+      if (option.value === undefined) {
+        return;
+      }
+      parsedOptions.push({ text: option.value });
+    });
+
+    return parsedOptions;
```

Comment:
Thank you! I didn't expect `flatMap` to work (in the sense of not causing type errors), since `flat` and `map` together don't. I do prefer it too, I'm gonna change it.